Deribit/DeribitLogic/Services/DeribitService.cs

```diff
@@ -0,0 +1,154 @@
+
+        string jsonString = JsonSerializer.Serialize(message);
+
+        await _clientWebSocketWrapper.SendMessageAsync(jsonString, WebSocketMessageType.Text, CancellationToken.None);
+
+        string result = await _clientWebSocketWrapper.ReceiveMessageAsync(CancellationToken.None);
+
+        //todo: less magic number, use own exception type
+        if (result.Contains("result") == false
+            || result.Contains("error") == true) 
+        {
+            throw new Exception("Authentication error!");
+        }
+
+        _logger.LogInformation("Authentication was successful!");
+    }
+
+    public async Task ConnectAsync()
+    {
+        string url = getUrlBasedOnConfig();
+
+        await _clientWebSocketWrapper.ConnectAsync(url, CancellationToken.None);
+    }
+
+    public async Task DisconnectAsync()
+    {
+        _isRun = false;
+        await SubscribeAsync(_unsubscribe);
+        await _clientWebSocketWrapper.DisonnectAndDisposeAsync(WebSocketCloseStatus.NormalClosure, CancellationToken.None);
+    }
+
+    public async Task InitializeAsync()
+    {
+        await ConnectAsync();
+        await AuthenticateAsync();
+        await SubscribeAsync(_subscribe);
+    }
+
+    //TODO: add logging and exception handling
+    public async Task RunListener()
+    {
+        while (_isRun)
+        {
+            string result = await _clientWebSocketWrapper.ReceiveMessageAsync(CancellationToken.None);
+            _logger.LogInformation($"Received: {result}");
+        }
+    }
+
+    public async Task SubscribeAsync(string subscribeMethod)
+    {
+        var config = _appSettings.DeribitApiClientConfig; //todo: put global variable
+
+        Channel channel = new Channel()
+        {
+            Channels = config.SubscribeTo
+        };
+
+        Message<Channel> message = new Message<Channel>() { Method = subscribeMethod, Params = channel }; //todo: put common method
+
+        string jsonString = JsonSerializer.Serialize(message);
+
+        await _clientWebSocketWrapper.SendMessageAsync(jsonString, WebSocketMessageType.Text, CancellationToken.None);
+
+        int cnt = 0;
+        bool isFine = false;
+
+        //todo: find better solution for the sporadic behaviour
+        while (cnt < 3 || isFine == false)
+        {
+            string result = await _clientWebSocketWrapper.ReceiveMessageAsync(CancellationToken.None);
+
+            //todo: less magic number, add own exception type
+            if (result.Contains("result") == false
+                || result.Contains("error") == true)
+            {
+                cnt++;
+            }
+            else
+            {
+                isFine = true;
+                break;
+            }
+        }
+
+        if (isFine == false)
+        {
+            throw new Exception("Subscription error!");
+        }
+
+        _logger.LogInformation("Subscription/Unsubscription was successful!");
+    }
+
+    //TODO: make it async
```

Comment:
Semmi szükség ezt async-á tenni. Mivel az egyetlen bemenete a config, amit konstruktorban kap meg, így akár ezt a számítást is 1x elég lenne elvégezni és ráadásul semmi IO vagy többszálúság nincs benne, ami miatt kéne az async.